os/drivers/audio/alc5658.c

```diff
@@ -729,41 +748,34 @@ static int alc5658_start(FAR struct audio_lowerhalf_s *dev)
 {
 
 	FAR struct alc5658_dev_s *priv = (FAR struct alc5658_dev_s *)dev;
-	int entry;
-	if (priv->running) {
-		return OK;
-	}
-
 	audvdbg(" alc5658_start Entry\n");
-	alc5658_exec_i2c_script(priv, codec_init_inout_script1, sizeof(codec_init_inout_script1) / sizeof(t_codec_init_script_entry));
+	alc5658_takesem(&priv->devsem);
 
-	entry = alc5658_get_sample_rate_script(priv->samprate);
-	if (entry == -1) {
-		return -EINVAL;
+	if (priv->running) {
+		goto alcstart_ok;
 	}
 
-	alc5658_exec_i2c_script(priv, g_sample_entry[entry].script, g_sample_entry[entry].size / sizeof(t_codec_init_script_entry));
-	alc5658_exec_i2c_script(priv, codec_init_inout_script2, sizeof(codec_init_inout_script2) / sizeof(t_codec_init_script_entry));
+	/* Register cb for io error */
+	I2S_ERR_CB_REG(priv->i2s, alc5658_io_err_cb, priv);
```

Comment:
Other than overflow and underflow, there can be a i2s timeout error also. I think that also needs to be handled.

Reply:
return value of enqueue request would address timeout error.

Reply:
No. Its not handled by enqueue request. 
Timeout happens much later after enqueue. It is indicated by result parameter of alc5658_rxtxcallback. But its being ignored.

Reply:
I think its better to just print an error message in alc5658_rxtxcallback when timeout error happens.

Reply:
Agreed